matter_server/server/ota/dcl.py

```diff
@@ -0,0 +1,120 @@
+"""Handle OTA software version endpoints of the DCL."""
+
+from http import HTTPStatus
+import logging
+from typing import Any, cast
+
+from aiohttp import ClientError, ClientSession
+
+from matter_server.common.errors import UpdateCheckError
+from matter_server.server.helpers import DCL_PRODUCTION_URL
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def _get_software_versions(vid: int, pid: int) -> Any:
+    """Check DCL if there are updates available for a particular node."""
+    async with ClientSession(raise_for_status=False) as http_session:
```

Comment:
So, we creating a new http session each time ? Why not use a shared http session at server level ?

Reply:
Good point! I've moved this up, so we only create one `ClientSession` per update check. But we probably could move it even further up, even make a shared client session for the whole server as you suggest. But if we do that, we should reuse this session in other places too, like vendor information and PAA certificate fetching. I'd rather prefer to do this in a follow up.

Reply:
Sure, it was just an observeration - write it somewhere down as follow up improvements. 
Not needed for MVP now.